src/compiler/types.ts

```diff
@@ -8822,6 +8822,7 @@ namespace ts {
         readonly includeInlayPropertyDeclarationTypeHints?: boolean;
         readonly includeInlayFunctionLikeReturnTypeHints?: boolean;
         readonly includeInlayEnumMemberValueHints?: boolean;
+        readonly allowRenameOfImportPath?: boolean;

```

Comment:
Ironically, I think this always belonged in `UserPreferences`, and `providePrefixAndSuffixTextForRename` always should have been a per-request option. 🙃 Baby steps in the right direction...